vtrace/platforms/linux.py

```diff
@@ -307,18 +307,27 @@ def platformExec(self, cmdline):
         self.execing = True
         cmdlist = e_cli.splitargs(cmdline)
         os.stat(cmdlist[0])
-        pid = os.fork()
-
+        
+        pid = os.fork() 
+        
+        # Child will be blocked after execv because of trace me!
         if pid == 0:
             v_posix.ptrace(v_posix.PT_TRACE_ME, 0, 0, 0)
-            # Make sure our parent gets some cycles
-            time.sleep(0.1)
             os.execv(cmdlist[0], cmdlist)
             sys.exit(-1)
 
-        if v_posix.ptrace(PT_ATTACH, pid, 0, 0) != 0:
-            raise Exception("PT_ATTACH failed! linux platformExec")
-
+        # Parent will wait for the child
+        for i in xrange(0, 33):
```

Comment:
In theory it can and in practice it does work without it. But since it has
the WNOHANG flag set I decided to give some chance of the system if
the first call fails. Since I am not 100% sure the kernel executes the
child first.

2015-04-26 21:49 GMT+02:00 invisig0th <notifications@github.com>:

> In vtrace/platforms/linux.py
> <https://github.com/vivisect/vivisect/pull/60#discussion_r29113120>:
>
> >              os.execv(cmdlist[0], cmdlist)
> >              sys.exit(-1)
> >
> > -        if v_posix.ptrace(PT_ATTACH, pid, 0, 0) != 0:
> > -            raise Exception("PT_ATTACH failed! linux platformExec")
> > -
> > +        # Parent will wait for the child
> > +        for i in xrange(0, 33):
>
> looks like a great fix overall ( i see now that it had a race condition
> that it doesn't need to :) what's up with the 33 loop here? is it required,
> or could we possibly do without it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vivisect/vivisect/pull/60/files#r29113120>.
>